Networking/Networking/Remote/WCAnalyticsCustomerRemote.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+public class WCAnalyticsCustomerRemote: Remote {
+
+    /// Retrieves the `Customer`collection from `/wc-analytics/customers`
+    ///
+    /// - Parameters:
+    ///     - siteID: Site for which we'll fetch the customers.
+    ///     - completion: Closure to be executed upon completion.
+    ///
+    func retrieveCustomers(for siteID: Int64, completion: @escaping (Result<[WCAnalyticsCustomer], Error>) -> Void) {
+        let path = "customers"
+        let request = JetpackRequest(
+            wooApiVersion: .wcAnalytics,
+            method: .get,
+            siteID: siteID,
+            path: path,
+            parameters: nil
+        )
+
+        let mapper = WCAnalyticsCustomerMapper(siteID: siteID)
+        enqueue(request, mapper: mapper, completion: completion)
+    }
+
+    /// Retrieves a `Customer` collection from `/wc-analytics/customers` based on the `?search=` parameter
+    ///
+    /// - Parameters:
+    ///     - siteID: Site for which we'll fetch the customer.
+    ///     - name: Name of the customer that will be retrieved
+    ///     - completion: Closure to be executed upon completion.
+    ///
+    func retrieveCustomerByName(for siteID: Int64, with name: String, completion: @escaping (Result<[WCAnalyticsCustomer], Error>) -> Void) {
```

Comment:
I'm assuming we may have more than one customer with the same name ( specially once we add support for partial search, we'll get more results than one match ), that's why for now I'm returning an `[WCAnalyticsCustomer]` entity here.